lib/fh_cluster.js

```diff
@@ -49,21 +134,27 @@ function setOnWorkerListeningHandler() {
     var host = address.address || address.addressType === 4 ? '0.0.0.0' : '::';
     var addr = host + ':' + address.port;
     console.log('Cluster worker',  worker.id, 'is now listening at', addr);
+
+    //A new process is listening, need to check if any bound tasks need to be assigned.
+    boundTasks.setUpBoundTasks(singleWorkerTasks, cluster);
+  });
+
+  cluster.on('exit', function(){
+    //If a worker exits, check that all bound workers are on a running worker.
+    boundTasks.setUpBoundTasks(singleWorkerTasks, cluster);
   });
 }
 
 function setupOnExitBackOff(strategy, backoffResetTimeout) {
-  cluster.on('disconnect', function(worker) {
-    if (cluster.isMaster) {
-      var nextRetry = strategy.next();
-      console.log('Worker #', worker.id, 'disconnected. Will retry in',
-                  nextRetry, 'ms');
-
-      setTimeout(function() {
-        cluster.fork();
-        resetBackoffResetTimeout(strategy, backoffResetTimeout);
-      }, nextRetry);
-    }
+  cluster.on('exit', function(worker) {
```

Comment:
I know we discussed this before, but I've forgotten (it's probably good to have details here also for posterity) -- why exit instead of disconnect?